File: app/src/main/java/com/belous/v/clrc/other/di/AppComponent.java
package com.belous.v.clrc.other.di;


import com.belous.v.clrc.view.dialog.BaseDialog;
import com.belous.v.clrc.view.dialog.FoundDialog;
import com.belous.v.clrc.view.fragment.InfoFragment;
import com.belous.v.clrc.view.fragment.ListFragment;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(FoundDialog foundDialog);

    void inject(BaseDialog baseDialog);

    void inject(ListFragment listFragment);

    void inject(InfoFragment infoFragment);
}

File: app/src/main/java/com/belous/v/clrc/di/AppComponent.kt
package com.belous.v.clrc.di

import com.belous.v.clrc.ui.feature_main.MainFragment
import com.belous.v.clrc.ui.feature_yeelight.YeelightFragment
import com.belous.v.clrc.MainActivity
import com.belous.v.clrc.ui.dialog.FoundDialog
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {

    fun inject(mainActivity: MainActivity)

    fun inject(foundDialog: FoundDialog)

    fun inject(mainFragment: MainFragment)

    fun inject(yeelightFragment: YeelightFragment)
}